chiselFrontend/src/main/scala/Chisel/Assert.scala

```diff
@@ -28,32 +29,36 @@ object assert { // scalastyle:ignore object.name
     * defs need to be compiled first and the SBT project is not set up to do
     * that
     */
-  def apply(cond: Bool, message: String): Unit = macro apply_impl_msg
-  def apply(cond: Bool): Unit = macro apply_impl  // macros currently can't take default arguments
+  // Macros currently can't take default arguments, so we need two functions to emulate defaults.
+  def apply(cond: Bool, message: String): Unit = macro AssertTransform.apply_msg
+  def apply(cond: Bool): Unit = macro AssertTransform.apply
 
-  def apply_impl_msg(c: Context)(cond: c.Tree, message: c.Tree): c.Tree = {
+  def do_apply(cond: Bool, message: String)(sourceInfo: SourceInfo): Unit = macro apply_impl_msg
+  def do_apply(cond: Bool)(sourceInfo: SourceInfo): Unit = macro apply_impl
+
+  def apply_impl_msg(c: Context)(cond: c.Tree, message: c.Tree)(sourceInfo: c.Tree): c.Tree = {
```

Comment:
Still is odd to plum the same information into the same point. (And are thus susceptible to discrepancies.)

Besides, lineContent is deprecated so you are supposed to get rid of it.